fletcher/algorithms/string_builder.py

```diff
@@ -0,0 +1,168 @@
+import math
+from typing import List
+
+import numba
+import numpy as np
+import pyarrow as pa
+from cffi import FFI
+
+class ByteVector:
+    """
```

Comment:
Doesn't debugging without a JIT maybe work by setting `NUMBA_DISABLE_JIT` as environment variable?

Reply:
No, it failed. Some annotations caused errors due to missing fields and I don't think it may conceptionally work since it heavily uses C functions. The python debugger has no clue about those. In the CH Project I saw that Visual Studio (Pro) actually can debug python and C/C++ in one session do you know of any other development setup which can do this?

Reply:
On Linux `gdb` can deal with this if you install the right packages.